packages/web/integrations/notifi/notifi-subscription-card/fetched-card/history-rows.tsx

```diff
@@ -0,0 +1,432 @@
+import { NotifiFrontendClient } from "@notifi-network/notifi-frontend-client";
+import dayjs from "dayjs";
+import { useRouter } from "next/router";
+import { FunctionComponent, useCallback, useMemo } from "react";
+
+import { Icon } from "~/components/assets";
+
+import { useNotifiModalContext } from "../../notifi-modal-context";
+
+export type HistoryRowData = Awaited<
+  ReturnType<NotifiFrontendClient["getNotificationHistory"]>
+>["nodes"][number];
+
+export const HistoryRows: FunctionComponent<{
+  rows: ReadonlyArray<HistoryRowData>;
+}> = ({ rows }) => {
+  return (
+    <ul>
+      {rows.map((row, key) => {
+        return <HistoryRow key={key} row={row} />;
+      })}
+
+      {dummyRows.map((row, key) => (
+        <HistoryRow row={row} key={key} />
+      ))}
+    </ul>
+  );
+};
+
+interface RowProps {
+  row: HistoryRowData | DummyRow;
+  dummyRow?: DummyRow;
+}
+
+const validateHistoryRow = (
+  row: HistoryRowData | DummyRow
+): row is HistoryRowData => {
+  return row.__typename !== "DummyRow";
+};
+
+export const HistoryRow: FunctionComponent<RowProps> = ({ row }) => {
+  const { renderView, selectedHistoryEntry, setSelectedHistoryEntry } =
+    useNotifiModalContext();
+  const router = useRouter();
+
+  const { emoji, title, message, cta, timestamp, popOutUrl } = useMemo(() => {
+    if (row.__typename !== "DummyRow") {
+      const day = dayjs(row.createdDate);
+      const isToday = day.isAfter(dayjs(Date.now()).subtract(1, "day"));
+      const isYesterday =
+        day.isAfter(dayjs(Date.now()).subtract(2, "day")) && !isToday;
+      const timestamp = isToday
+        ? day.format("h:mm A")
+        : isYesterday
+        ? day.format("h:mm A") + " Yesterday"
+        : day.format("MMMM D, YYYY h:mm A");
+
+      const rowProps = {
+        emoji: "😵",
+        title: "Unsupported Notification",
```

Comment:
Will you guys have localizations? I'd add them to a "notifi" member of our translations file en.json. Then, you should be able to run `yarn machine-translate` to generate initial translations automatically.

Reply:
Thanks! Will use it with some constant values.
In this case of `rowProps`, It is dynamic value though. we might not be able to adopt in this block.

Reply:
Oh ... It means that in this case the `rowProps`  content changes over time. 
each rendered component has its own corresponding content (`rowProps`).

Reply:
And how does that prevent localization? I'd like to, if possible, avoid having unlocalized content along with localized content